deploy-nebula-graph-with-docker-and-compose.md

```diff
@@ -0,0 +1,173 @@
+# Deploying Nebula Graph Cluster with Docker and Docker Compose
+
+## Overview
+
+In this document, we will walk you through the process of deploying a **Nebula Graph** cluster with [Docker](https://docs.docker.com/install/) and [Docker Compose](https://docs.docker.com/compose/install/). We will also show you how to [check the services status of **Nebula Graph**](#checking-the-status-and-ports-of-nebula-graph-services), how to [check the cluster data and logs](#checking-the-cluster-data-and-logs), and how to [stop the services of **Nebula Graph**](#stopping-the-services-of-nebula-graph).
+
+## Prerequisite
+
+Before you start deploying the **Nebula Graph** cluster, ensure that you have installed the latest version of [Docker](https://docs.docker.com/install/) and [Docker Compose](https://docs.docker.com/compose/install/).
+
+## Services of Nebula Graph to Be Deployed
+
+In this guide, we are going to deploy the following services of **Nebula Graph**:
+
+* 3 replicas of nebula-metad service
+* 3 replicas of nebula-storaged service
+* 1 replica of nebula-graphd service
+
+## Deploying the Nebula Graph Cluster
+
+You can deploy the **Nebula Graph** cluster by the following steps:
+
+1. Clone the `nebula-docker-compose` repository to your local computer.
+
+```shell
+$ git clone https://github.com/vesoft-inc/nebula-docker-compose.git
+```
+
+2. Change your current directory to the `nebula-docker-compose` directory.
+
+```shell
+$ cd nebula-docker-compose/
+```
+
+3. Start all the services of **Nebula Graph**.
+
+```shell
+$ sudo docker-compose up -d
+```
+
+The following information is displayed:
+
+```$
+Creating nebula-docker-compose_metad2_1 ... done
+Creating nebula-docker-compose_metad1_1 ... done
+Creating nebula-docker-compose_metad0_1 ... done
+Creating nebula-docker-compose_storaged2_1 ... done
+Creating nebula-docker-compose_graphd_1    ... done
+Creating nebula-docker-compose_storaged0_1 ... done
+Creating nebula-docker-compose_storaged1_1 ... done
+```
+
+4. Pull the `vesoft/nebula-console:nightly` image to your local computer.
+
+```shell
+$ docker pull vesoft/nebula-console:nightly
+```
+
+**Note**: 
+
+a. We will use the `nebula-console` docker container to connect to the graph service of **Nebula Graph**.
+
+b. If you have pulled the `vesoft/nebula-console` image before, remove it with the following command before you pull it again:
+
+   * `docker rm $(docker ps -qa -f status=exited)`
+   * `docker rmi vesoft/nebula-console:nightly`
+   
+c. If you have pulled the **Nebula Graph** images before, you can update the images with the following command:
+
+```
+$ docker-compose pull
+```
+
+5. Connect to the graph service of **Nebula Graph**.
+
+```shell
+$ sudo docker run --rm -ti --network=host vesoft/nebula-console:nightly --addr=127.0.0.1 --port=3699
```

Comment:
You can tell user how to add a docker group like [this document](https://docs.docker.com/install/linux/linux-postinstall/) rather than use `sudo` permission in each docker command.

Reply:
If the user does not try your recommendation, what will you do?

Even in the Docker Documentation, you can find tons of sudo